packages/ui/src/TextFieldFormik.tsx

```diff
@@ -1,11 +1,11 @@
 import React, { ReactElement } from 'react'
 import { useField } from 'formik'
 
-import { TextField, TextFieldExtraProps } from './TextField'
+import { TextField, TextFieldExtraProps, TextFieldTypes } from './TextField'
 import { FieldValidatorGeneric, getFieldError } from './utils/formik'
 import { ExtractKeysOfValueType } from './utils/types'
 
-export type TextFieldFormikProps<V extends object> = Exclude<TextFieldExtraProps<'text'>, 'type'> & {
+export type TextFieldFormikProps<V extends object> = TextFieldExtraProps<TextFieldTypes> & {
```

Comment:
We should not allow `number` as the type. I think we should allow `'email' | 'search' | 'tel' | 'text' | 'url' | undefined` only.

BTW, we should pass the proper type on L22

Reply:
@aigoncharov Wow, good catch! But why shouldn’t we allow `number` type?

Reply:
@jozefhruska we have `NumberField` for it, right? And for `password` we have `PasswordField`.

Reply:
Cool, now I understand 👌 Will make the change, thanks.